yarn-project/simulator/src/avm/avm_simulator.test.ts

```diff
@@ -560,11 +605,13 @@ describe('AVM simulator: transpiled Noir contracts', () => {
       const context = initContext();
       jest
         .spyOn(context.persistableState.hostStorage.contractsDb, 'getBytecode')
-        .mockReturnValueOnce(Promise.resolve(nestedBytecode));
+        .mockReturnValue(Promise.resolve(nestedBytecode));
 
       const results = await new AvmSimulator(context).executeBytecode(callBytecode);
 
       expect(results.reverted).toBe(true); // The outer call should revert.
+      // TODO(fcarreiro): revertReason lost in translation between results.
+      // expect(results.revertReason).toEqual(/StaticCallStorageAlterError/);
     });
```

Comment:
Why is the revertReason lost? Also is there a ticket or more info on how/when this will be solved?

Reply:
I think it's because the way nested calls in PublicExecution* are handled. Nested executions in current public are not expected to revert and carry on, they would throw and everything dies. This should certainly work once we remove the usage of the ACVM, but I might need to get it working somehow before that.